src/Illuminate/Database/Eloquent/SoftDeletingScope.php

```diff
@@ -104,6 +106,8 @@ protected function addWithoutTrashed(Builder $builder)
 
             $builder->withoutGlobalScope($this)->whereNull(
                 $model->getQualifiedDeletedAtColumn()
+            )->orWhere(
+                $model->getQualifiedDeletedAtColumn(), '>', now()
```

Comment:
This scope needs to be parenthesis-wrapped so it doesn't leak as an operand in subsequent scopes.

```php
$builder->withoutGlobalScope($this)->withoutGlobalScope($this)->where(function ($query) use (model) {
    $query->whereNull(
        $model->getQualifiedDeletedAtColumn()
    )->orWhere(
        $model->getQualifiedDeletedAtColumn(), '>', now()
    );
});
```

With the current PR, the below unverified users query would mistakenly include `$userActive`.

```php
$userActive = App\Models\User::factory()->create([
    'email_verified_at' => null,
    'deleted_at' => null,
]);

$userFutureDeleted = App\Models\User::factory()->create([
    'email_verified_at' => null,
    'deleted_at' => now()->addWeek(),
]);

App\Models\User::withoutTrashed()->whereNotNull('email_verified_at')->get();
```

Ordering scopes in the opposite order, `$userFutureDeleted` is included.

```php
App\Models\User::whereNotNull('email_verified_at')->withoutTrashed()->get();
```

Both queries should be empty for two unverified users.

```sql
SELECT * FROM users
WHERE email_verified_at IS NOT NULL
  AND deleted_at IS NULL
   OR deleted_at > ?
```

needs to be:

```sql
SELECT * FROM users
WHERE email_verified_at IS NOT NULL
  AND (deleted_at IS NULL OR deleted_at > ?)
```

Reply:
Fixed in last commit.